pkg/api/annotation_key_constants.go

```diff
@@ -47,6 +47,7 @@ const (
 
 	// CreatedByAnnotation represents the key used to store the spec(json)
 	// used to create the resource.
+	// This field is deprecated in favor of ControllerRef.
```

Comment:
Please add `(see #44407)` at the end so people can find out the context.

Also we usually file a GitHub issue to track the removal step, and add a TODO like:

```go
// TODO(#removal-issue-num): Remove this in v1.9.
```